hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupManifest.java

```diff
@@ -439,7 +440,7 @@ public BackupManifest(FileSystem fs, Path backupPath) throws BackupException {
           } catch (Exception e) {
             throw new BackupException(e);
           }
-          this.backupImage = BackupImage.fromProto(proto);
+          this.backupImage = hydrateRootDir(BackupImage.fromProto(proto), backupPath);
```

Comment:
This is the meat of the change. The other files are either tests, or some refactoring to make testing easier. I've deployed this to a test environment and validated that we can run a restore of a backup, even after it is replicated to a secondary location.